ndc/ndc.go

```diff
@@ -0,0 +1,188 @@
+	//NDC does not have dashes, we can't do conversion
+	if len(parts) < 3 {
+		return ndc
+	}
+	//NDC already has 11 digits
+	if len(SanitizeNDC(ndc)) != 10 {
+		return ndc
+	}
+
+	//first part could be a candidate
+	candidate := parts[0]
+	if len(candidate) == 4 {
+		parts[0] = addLeadingZero(candidate)
+		return strings.Join(parts, "")
+	}
+
+	//second part could be a candidate
+	candidate = parts[1]
+	if len(candidate) == 3 {
+		parts[1] = addLeadingZero(candidate)
+		return strings.Join(parts, "")
+	}
+
+	//last part could be a candidate
+	candidate = parts[2]
+	if len(candidate) == 1 {
+		parts[2] = addLeadingZero(candidate)
+		return strings.Join(parts, "")
+	}
+
+	return strings.Replace(ndc, "-", "", -1)
+}
+
+//ConvertNDCTo10Digits returns ndc in 10 digits
+//11 digit NDC code is 5-4-2. Could be written as 4-4-2 or 5-3-2 or 5-4-1
+func ConvertNDCTo10Digits(ndc string) string {
+	parts := strings.Split(ndc, "-")
+	if len(parts) < 3 {
+		return SanitizeNDC(ndc)
+	}
+
+	//first part could be a candidate
+	candidate := parts[0]
+	if len(candidate) == 5 && isLeadingZero(candidate) {
+		parts[0] = removeLeadingZero(candidate)
+		return strings.Join(parts, "")
+	}
+
+	//second part could be a candidate
+	candidate = parts[1]
+	if len(candidate) == 4 && isLeadingZero(candidate) {
+		parts[1] = removeLeadingZero(candidate)
+		return strings.Join(parts, "")
+	}
+
+	//last part could be a candidate
+	candidate = parts[2]
+	if len(candidate) == 2 && isLeadingZero(candidate) {
+		parts[2] = removeLeadingZero(candidate)
+		return strings.Join(parts, "")
+	}
+
+	return strings.Replace(ndc, "-", "", -1)
+}
+
+//SanitizeNDC cleans up the ndc code
+func SanitizeNDC(ndc string) string {
+	sanitized := strings.Replace(ndc, "-", "", -1)
+	return strings.TrimSpace(sanitized)
+}
+
+//AddDashesTo11DigitNDC adds the dash to format the NDC
+func AddDashesTo11DigitNDC(s string) string {
+	if len(s) < 11 {
+		return s
+	}
+
+	return fmt.Sprintf("%s-%s-%s", s[:5], s[5:9], s[9:])
+}
+
+//AddDashesTo10DigitNDC adds the dash to format the NDC
+func AddDashesTo10DigitNDC(s string) string {
+	if len(s) != 10 {
+		return s
+	}
+
+	// if first character is 0 add 0 in first part eg. 00XXX-XXXX-XX
+	if s[0] == '0' {
+		return fmt.Sprintf("0%s-%s-%s", s[:4], s[4:8], s[8:])
+	}
+
+	// if first character is not 0 add 0 in second part eg. XXXXX-0XXX-XX
+	return fmt.Sprintf("%s-0%s-%s", s[:5], s[5:8], s[8:])
+}
+
+// NDCFromZohoSKU returns NDC from SKU with format X10-digit-NDCY
+func NDCFromZohoSKU(formattedSKu string) string {
+	sku := strings.ReplaceAll(formattedSKu, "-", "")
+	skuLength := len(sku)
+
+	// if SKU format is not based on the agreed spec
+	if skuLength != 12 {
+		return formattedSKu
+	}
+
+	// remove first and last charatcer from sku
+	sku = string(sku[1 : skuLength-1])
+
+	return AddDashesTo10DigitNDC(sku)
+}
+
+func isLeadingZero(part string) bool {
+	return part[0] == '0'
+}
+
+func removeLeadingZero(part string) string {
+	return part[1:]
```

Comment:
Maybe we can add check if it is 0 before removing it?

Reply:
That is checked by `isLeadingZero` function

Reply:
@Swechhya could we use strings.TrimPrefix() here, instead of using isLeadingZero check and removeLeadingZero. I think strings.TrimPrefix() does the job of both the function. Checkout the examples here: https://pkg.go.dev/strings#TrimPrefix

Reply:
@DibeshMSShrestha  `strings.TrimPrefix()` does the same thing but these functions are used in `ConvertNDCTo10Digits` and here we need to check if that part of the NDC has leading zero and remove the zero and then return from the function, otherwise we need to check the second part. So we cannot use  `strings.TrimPrefix()` function here.